### ISSUE TITLE
✨ Add 'Publishing group' filter to 'Pools' table

### ISSUE THREAD

**Issue by Jerryescandon** [GCTC-NTGC/gc-digital-talent#6169]
## ✨ Feature

Include a filter on the pools table to be able to sort by "Publishing group"

## 🕵️ Details

DCM wants to be able to quickly get to their Continues Intake pools, which should be under their own publishing group. Adding this filter will allow for easier access to these pools.

## 📸 Screenshot

Pools page

![image](https://user-images.githubusercontent.com/16977254/229599808-5558b9b0-5b89-45f7-a9ca-276f716961c2.png)

Pools filter modal

![image](https://user-images.githubusercontent.com/16977254/229601436-3ec976be-850d-45c4-8ab5-4523032ccf8c.png)

## ✅ Acceptance Criteria

- [ ] Pools table filterable on publishing group
- [ ] Works in EN and FR

## 🛑 
```[tasklist]
### Blocked by
- [ ] #9718 
```

**Comment by vd1992** [comment]
the pools table isn't set up like the pool candidate or users table, so added filtering won't be able to match them easily
(pool candidate and users table use the api-driven table, pools table does not)

**Comment by petertgiles** [comment]
Note from Eric: https://react-table-v7.tanstack.com/docs/api/useFilters

**Comment by esizer** [comment]
> the pools table isn't set up like the pool candidate or users table, so added filtering won't be able to match them easily (pool candidate and users table use the api-driven table, pools table does not)
> 
> so adding filtering should stay a separate task

Yeah, if we want to keep this as a client side table we may need to look at filtering. It may be a little tricky since the client side tables use the search dropdown for filters which would override any filters in the dialog (how it currently works) So it may require a bit of refactoring of the base table component to see how we can prevent the search from overriding other filters.

Or, we could create a paginated query and convert it to a server side table (_could_ be more beneficial in the long run).

--- 

Also, there is some work being done on the filter dialogs that will be impacting this so I an going to add #8776 as a blocker.

**Comment by mnigh** [comment]
> Or, we could create a paginated query and convert it to a server side table (_could_ be more beneficial in the long run).

- @Jerryescandon @tristan-orourke is this feature still something the client is interested in? if so, should we create a paginated query and convert it to a server side table as mentioned by eric?

**Comment by tristan-orourke** [comment]
You know what, it's probably time we make the pool query a paginated query, anyway.
